### PR TITLE
Fix the example in README.md

### PR DESCRIPTION
**Pull request by jonas089**
The example in README.md is incorrect, I attempted to fix it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,19 +65,37 @@ use pgp::{SignedSecretKey, Message, Deserializable};
...
 
 // Verify this message
-msg.verify(&pkey).unwrap();
+msg.verify(&secret_key.public_key()).unwrap();
 
 let msg_content = msg.get_content().unwrap(); // actual message content
 ```
 
+### Generate and Verify a Signature with a GPG Keypair
+```rust
```
Comment (hko-s) [inline README.md:81]:
````suggestion
```rust
use std::fs;
use pgp::{Deserializable, SignedPublicKey, SignedSecretKey};
use pgp::types::{PublicKeyTrait, SecretKeyTrait};
use pgp::crypto::hash::HashAlgorithm;

````

File: README.md [hunk 2]
```diff
@@ -65,19 +65,37 @@ use pgp::{SignedSecretKey, Message, Deserializable};
...
+let key_file = "key.sec.asc";
+let pub_key_file = "key.asc";
+
+let secret_key_string = fs::read_to_string(key_file).expect("Failed to load secret key");
+let signed_secret_key = SignedSecretKey::from_string(&secret_key_string).unwrap().0;
+
+let data: Vec<u8> = vec![0u8];
+
+// create a new signature
+let new_signature = signed_secret_key.create_signature(|| "".to_string(), pgp::crypto::hash::HashAlgorithm::MD5, &data).unwrap();
```
Comment (hko-s) [inline README.md:91]:
```suggestion
let new_signature = signed_secret_key.create_signature(|| "".to_string(), HashAlgorithm::default(), &data).unwrap(); 
```

File: README.md [hunk 3]
```diff
@@ -65,19 +65,37 @@ use pgp::{SignedSecretKey, Message, Deserializable};
...
+
+let data: Vec<u8> = vec![0u8];
+
+// create a new signature
+let new_signature = signed_secret_key.create_signature(|| "".to_string(), pgp::crypto::hash::HashAlgorithm::MD5, &data).unwrap(); 
+
+let key_string = fs::read_to_string(pub_key_file).expect("Failed to load public key");
+let public_key = SignedPublicKey::from_string(&key_string).unwrap().0;
+
+public_key.verify_signature(pgp::crypto::hash::HashAlgorithm::MD5, &data, &new_signature).unwrap();
```
Comment (hko-s) [inline README.md:96]:
```suggestion
public_key.verify_signature(HashAlgorithm::default(), &data, &new_signature).unwrap();
```

File: README.md (hunk 2 above)
Comment (hko-s) [reply README.md:91]: MD5 is not appropriate for cryptographic use anymore (see e.g. [here](https://en.wikipedia.org/wiki/MD5#Overview_of_security_issues)).

`HashAlgorithm::default()` returns a current hash algorithm.

**Review by hko-s — COMMENTED** [review]
Hey! Thanks for checking and improving the example and adding another one!
I added some minor suggestions to clarify terminology and use a cryptographic hash algorithm. I think github will allow you to one-click accept the diffs, if you agree with them :)

**Comment by jonas089** [comment]
lgtm :)

**Review by dignifiedquire — APPROVED** [review]
thanks

**Pull request merged**